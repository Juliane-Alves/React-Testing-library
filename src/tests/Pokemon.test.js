import React from 'react';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from '../App';
import renderWithRouter from '../helpers/renderWithRouter';

describe('test se as cartas dos pokemons estão renderizando corretamente', () => {
  test('Test se a carta do pokemon renderiza', () => {
    renderWithRouter(<App />);
    const title = screen.getByRole('heading', { name: /encountered pokémons/i });
    expect(title).toBeInTheDocument();

    const moreDetailsLink = screen.getByRole('link', { name: /more details/i });
    userEvent.click(moreDetailsLink);
    const pokemonfav = screen.getByText(/pokémon favoritado/i);
    expect(pokemonfav).toBeInTheDocument();

    const favCheck = screen.getByRole('checkbox', { name: /pokémon favoritado\?/i });
    userEvent.click(favCheck);

    const favoritePokemonsPage = screen.getByRole('link', { name: /favorite pokémons/i });
    userEvent.click(favoritePokemonsPage);

    const name = screen.getByText(/pikachu/i);
    const type = screen.getByText(/electric/i);
    const size = screen.getByText(/average weight:/i);
    expect(name).toBeInTheDocument();
    expect(type).toBeInTheDocument();
    expect(size).toBeInTheDocument();
  });
  test('Test se o pokémon correto é renderizado', () => {
    renderWithRouter(<App />);
    const pokePoison = screen.getByRole('button', { name: /poison/i });
    userEvent.click(pokePoison);

    const pokeEkans = screen.getByText(/ekans/i);
    expect(pokeEkans).toBeInTheDocument();

    const moreDetailsLink = screen.getByRole('link', { name: /more details/i });
    userEvent.click(moreDetailsLink);

    const pokemonfav = screen.getByText(/pokémon favoritado\?/i);
    expect(pokemonfav).toBeInTheDocument();

    const favCheck = screen.getByRole('checkbox', { name: /pokémon favoritado/i });
    userEvent.click(favCheck);

    const favoritePokemonsPage = screen.getByRole('link', { name: /favorite pokémons/i });
    userEvent.click(favoritePokemonsPage);

    const pokeEk = screen.getByText(/ekans/i);
    expect(pokeEk).toBeInTheDocument();
  });
  test('Test se o tamanho do pokemon renderiza com os atributos correntos', () => {
    const { container } = renderWithRouter(<App
      measurementUnit="kg"
      value="6.0"
      averageWeight="Average weight"
    />);
    expect(container.firstChild).toMatchSnapshot();
    // Container(firstCHild) contém o container renderizado.
    // Sobre os testes que precisam de verificar os atributos, estava tentando fazer com o ToHaveAtributte e não funcionou,
  /*  obtive orientação de Nicole Calderari sobre o metodo Snapshot que vai criar uma pasta "por baixo dos panos"
    com o componente que estou usando e que vai conter as props, então nesse test se espera que os atributos estejam iguais
    ao Snapshot */
    // snapshot foi usado em todos os tests que rerifica atributo//
  });
  test('test se a imagem é renderizada com os atributos corretos', () => {
    renderWithRouter(<App />);
    const url = 'https://cdn2.bulbagarden.net/upload/b/b2/Spr_5b_025_m.png';
    const image = screen.getByRole('img', { name: /pikachu sprite/i });
    expect(image).toHaveAttribute('src', url);
  });
});

describe('testando a URL da pagina Pokedex', () => {
  test('test se contém um link de navegação "Mostrar detalhes"', () => {
    renderWithRouter(<App />);
    const link = screen.getByText(/more details/i);
    expect(link).toBeInTheDocument();
  });
  test('test se a url do link contém o id do pokemon', () => {
    const { container } = renderWithRouter(<App id="25" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
describe('Testando o link de navegação de um pokemon', () => {
  test('Test se redireciona para detalhes do pokemon quando é clicado', () => {
    renderWithRouter(<App />);
    const link = screen.getByText(/more details/i);
    userEvent.click(link);
    const pokeSummary = screen.getByRole('heading', { name: /summary/i });
    expect(pokeSummary).toBeInTheDocument();
  });
});

describe('testando o icone favorito', () => {
  test('Test se tem um icone de "fav" quando é adicionado aos favoritos', () => {
    const { container } = renderWithRouter(<App
      alt="Pikachu is marked as favorite"
      src="/star-icon.svg"
    />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test('test se o atributo contém o caminho "src" correto', () => {
    const { container } = renderWithRouter(<App src="/star-icon.svg" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test('test se contém um atributo alt com o nome do Pokemon', () => {
    const { container } = renderWithRouter(<App
      alt="Pikachu is marked as favorite"
    />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
