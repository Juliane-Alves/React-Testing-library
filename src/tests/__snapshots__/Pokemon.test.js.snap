// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test se as cartas dos pokemons estão renderizando corretamente Test se o tamanho do pokemon renderiza com os atributos correntos 1`] = `
<div
  class="App"
>
  <h1>
    Pokédex
  </h1>
  <nav>
    <a
      class="link"
      href="/"
    >
      Home
    </a>
    <a
      class="link"
      href="/about"
    >
      About
    </a>
    <a
      class="link"
      href="/favorites"
    >
      Favorite Pokémons
    </a>
  </nav>
  <div
    class="pokedex"
  >
    <h2>
      Encountered pokémons
    </h2>
    <div
      class="pokemon"
    >
      <div
        class="pokemon-overview"
      >
        <p
          data-testid="pokemon-name"
        >
          Pikachu
        </p>
        <p
          data-testid="pokemon-type"
        >
          Electric
        </p>
        <p
          data-testid="pokemon-weight"
        >
          Average weight: 
          6.0
           
          kg
        </p>
        <a
          href="/pokemons/25"
        >
          More details
        </a>
      </div>
      <img
        alt="Pikachu sprite"
        src="https://cdn2.bulbagarden.net/upload/b/b2/Spr_5b_025_m.png"
      />
      <img
        alt="Pikachu is marked as favorite"
        class="favorite-icon"
        src="/star-icon.svg"
      />
    </div>
    <div
      class="pokedex-buttons-panel"
    >
      <button
        class="button-text filter-button"
        data-testid=""
        type="button"
      >
        All
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Electric
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Fire
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Bug
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Poison
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Psychic
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Normal
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Dragon
      </button>
    </div>
    <button
      class="button-text pokedex-button"
      data-testid="next-pokemon"
      type="button"
    >
      Próximo pokémon
    </button>
  </div>
</div>
`;

exports[`testando a URL da pagina Pokedex test se a url do link contém o id do pokemon 1`] = `
<div
  class="App"
>
  <h1>
    Pokédex
  </h1>
  <nav>
    <a
      class="link"
      href="/"
    >
      Home
    </a>
    <a
      class="link"
      href="/about"
    >
      About
    </a>
    <a
      class="link"
      href="/favorites"
    >
      Favorite Pokémons
    </a>
  </nav>
  <div
    class="pokedex"
  >
    <h2>
      Encountered pokémons
    </h2>
    <div
      class="pokemon"
    >
      <div
        class="pokemon-overview"
      >
        <p
          data-testid="pokemon-name"
        >
          Pikachu
        </p>
        <p
          data-testid="pokemon-type"
        >
          Electric
        </p>
        <p
          data-testid="pokemon-weight"
        >
          Average weight: 
          6.0
           
          kg
        </p>
        <a
          href="/pokemons/25"
        >
          More details
        </a>
      </div>
      <img
        alt="Pikachu sprite"
        src="https://cdn2.bulbagarden.net/upload/b/b2/Spr_5b_025_m.png"
      />
      <img
        alt="Pikachu is marked as favorite"
        class="favorite-icon"
        src="/star-icon.svg"
      />
    </div>
    <div
      class="pokedex-buttons-panel"
    >
      <button
        class="button-text filter-button"
        data-testid=""
        type="button"
      >
        All
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Electric
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Fire
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Bug
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Poison
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Psychic
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Normal
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Dragon
      </button>
    </div>
    <button
      class="button-text pokedex-button"
      data-testid="next-pokemon"
      type="button"
    >
      Próximo pokémon
    </button>
  </div>
</div>
`;

exports[`testando o icone favorito Test se tem um icone de "fav" quando é adicionado aos favoritos 1`] = `
<div
  class="App"
>
  <h1>
    Pokédex
  </h1>
  <nav>
    <a
      class="link"
      href="/"
    >
      Home
    </a>
    <a
      class="link"
      href="/about"
    >
      About
    </a>
    <a
      class="link"
      href="/favorites"
    >
      Favorite Pokémons
    </a>
  </nav>
  <div
    class="pokedex"
  >
    <h2>
      Encountered pokémons
    </h2>
    <div
      class="pokemon"
    >
      <div
        class="pokemon-overview"
      >
        <p
          data-testid="pokemon-name"
        >
          Pikachu
        </p>
        <p
          data-testid="pokemon-type"
        >
          Electric
        </p>
        <p
          data-testid="pokemon-weight"
        >
          Average weight: 
          6.0
           
          kg
        </p>
        <a
          href="/pokemons/25"
        >
          More details
        </a>
      </div>
      <img
        alt="Pikachu sprite"
        src="https://cdn2.bulbagarden.net/upload/b/b2/Spr_5b_025_m.png"
      />
      <img
        alt="Pikachu is marked as favorite"
        class="favorite-icon"
        src="/star-icon.svg"
      />
    </div>
    <div
      class="pokedex-buttons-panel"
    >
      <button
        class="button-text filter-button"
        data-testid=""
        type="button"
      >
        All
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Electric
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Fire
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Bug
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Poison
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Psychic
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Normal
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Dragon
      </button>
    </div>
    <button
      class="button-text pokedex-button"
      data-testid="next-pokemon"
      type="button"
    >
      Próximo pokémon
    </button>
  </div>
</div>
`;

exports[`testando o icone favorito test se contém um atributo alt com o nome do Pokemon 1`] = `
<div
  class="App"
>
  <h1>
    Pokédex
  </h1>
  <nav>
    <a
      class="link"
      href="/"
    >
      Home
    </a>
    <a
      class="link"
      href="/about"
    >
      About
    </a>
    <a
      class="link"
      href="/favorites"
    >
      Favorite Pokémons
    </a>
  </nav>
  <div
    class="pokedex"
  >
    <h2>
      Encountered pokémons
    </h2>
    <div
      class="pokemon"
    >
      <div
        class="pokemon-overview"
      >
        <p
          data-testid="pokemon-name"
        >
          Pikachu
        </p>
        <p
          data-testid="pokemon-type"
        >
          Electric
        </p>
        <p
          data-testid="pokemon-weight"
        >
          Average weight: 
          6.0
           
          kg
        </p>
        <a
          href="/pokemons/25"
        >
          More details
        </a>
      </div>
      <img
        alt="Pikachu sprite"
        src="https://cdn2.bulbagarden.net/upload/b/b2/Spr_5b_025_m.png"
      />
      <img
        alt="Pikachu is marked as favorite"
        class="favorite-icon"
        src="/star-icon.svg"
      />
    </div>
    <div
      class="pokedex-buttons-panel"
    >
      <button
        class="button-text filter-button"
        data-testid=""
        type="button"
      >
        All
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Electric
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Fire
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Bug
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Poison
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Psychic
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Normal
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Dragon
      </button>
    </div>
    <button
      class="button-text pokedex-button"
      data-testid="next-pokemon"
      type="button"
    >
      Próximo pokémon
    </button>
  </div>
</div>
`;

exports[`testando o icone favorito test se o atributo contém o caminho "src" correto 1`] = `
<div
  class="App"
>
  <h1>
    Pokédex
  </h1>
  <nav>
    <a
      class="link"
      href="/"
    >
      Home
    </a>
    <a
      class="link"
      href="/about"
    >
      About
    </a>
    <a
      class="link"
      href="/favorites"
    >
      Favorite Pokémons
    </a>
  </nav>
  <div
    class="pokedex"
  >
    <h2>
      Encountered pokémons
    </h2>
    <div
      class="pokemon"
    >
      <div
        class="pokemon-overview"
      >
        <p
          data-testid="pokemon-name"
        >
          Pikachu
        </p>
        <p
          data-testid="pokemon-type"
        >
          Electric
        </p>
        <p
          data-testid="pokemon-weight"
        >
          Average weight: 
          6.0
           
          kg
        </p>
        <a
          href="/pokemons/25"
        >
          More details
        </a>
      </div>
      <img
        alt="Pikachu sprite"
        src="https://cdn2.bulbagarden.net/upload/b/b2/Spr_5b_025_m.png"
      />
      <img
        alt="Pikachu is marked as favorite"
        class="favorite-icon"
        src="/star-icon.svg"
      />
    </div>
    <div
      class="pokedex-buttons-panel"
    >
      <button
        class="button-text filter-button"
        data-testid=""
        type="button"
      >
        All
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Electric
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Fire
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Bug
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Poison
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Psychic
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Normal
      </button>
      <button
        class="button-text filter-button"
        data-testid="pokemon-type-button"
        type="button"
      >
        Dragon
      </button>
    </div>
    <button
      class="button-text pokedex-button"
      data-testid="next-pokemon"
      type="button"
    >
      Próximo pokémon
    </button>
  </div>
</div>
`;
